src/libutil/filesystem.cpp

```diff
@@ -494,22 +456,12 @@ Filesystem::temp_directory_path()
 std::string
 Filesystem::unique_path(string_view model)
 {
-#ifdef USE_BOOST_FILESYSTEM
-    // Boost filesystem has unique_path().
-    error_code ec;
-    filesystem::path p = filesystem::unique_path(u8path(model), ec);
-    return ec ? std::string() : pathstr(p);
-#else
     // std::filesystem does not have unique_path(). Punt!
-#    if defined(_WIN32)
-    // boost internally doesn't use MultiByteToWideChar (CP_UTF8,...
-    // to convert char* to wchar_t* because they do not know the encoding
-    // See boost/filesystem/path.hpp
-    // The only correct way to do this is to do the conversion ourselves
+#if defined(_WIN32)
     std::wstring modelStr = Strutil::utf8_to_utf16wstring(model);
-#    else
+#else
     std::string modelStr = model.str();
```

Comment:
This `modelStr` variable is not used (neither before nor after these changes). Is that a bug?

Reply:
I think that's an entirely separate bug! I'll make a separate PR for it since it's not C++17 specific.

Reply:
This topic moving to #4203